content/en/docs/concepts/extend-kubernetes/operator.md

```diff
@@ -0,0 +1,107 @@
+---
+title: Kubernetes operators
+reviewers:
+- erictune
+- lavalamp
+- cheftako
+- chenopis
+content_template: templates/concept
+weight: 30
+---
+
+{{% capture overview %}}
+
+Operators are software extensions to Kubernetes that make use of third-party
+resources to manage applications and their components. Operators follow
+Kubernetes priciples, notably the [control loop](/docs/concepts/#kubernetes-control-plane).
+
+{{% /capture %}}
+
+
+{{% capture body %}}
+
+## Motivation
+
+The Operator concept aims to capture the key aim of a human operator who
+is managing a service or set of services. Human operators who look after
+specific applications and services have deep knowledge of how the system
+ought to behave, how to deploy it, and how to react if there are problems.
+
+People who run workloads on Kubernetes often like to use automation to take
+care of repeatable tasks. The Operator pattern captures how you can write
+code to automate a task beyond what Kubernetes itself provides.
+
+## Operators in Kubernetes
+
+Kubernetes is designed for automation. Out of the box, you get lots of
+built-in automation from the core of Kubernetes. You can use Kubernetes
+to automate deploying and running workloads, *and* you can automate how
+Kubernetes does that.
+
+Kubernetes' {{< glossary_tooltip text="controllers" term_id="controller" >}}
+concept lets you extend the cluster's behaviour without modifying the code
+of Kubernetes itself.
+Operators are clients of the Kubernetes API that act as controllers for
```

Comment:
Is this true: Operators  ==  (CRDs and custom Controllers)?

Reply:
In pseudo-Rust I'd say that
```rust
impl <R,C> Operator<R, C> for SomeOperator where
    R: CustomResourceDefinition,
    C: Controller,
    (R+C): extends Kubernetes
```
(that last bit seems important)

Reply:
Hello @sftim. Yes, I think by definition the combo extends k8s.
- Check out the glossary reference, line 13, the link needs an opening paren?
- The `An example Operator` section is still a bit vague for me -- seems like pseudocode for writing an Operator. Could this content be combined with `Writing your own Op`?
- Do you need the link to the `What's next section` in line 108, that section is just below? 

Preview pages:
- https://deploy-preview-14458--kubernetes-io-master-staging.netlify.com/docs/concepts/extend-kubernetes/extend-cluster/#user-defined-types
- https://deploy-preview-14458--kubernetes-io-master-staging.netlify.com/docs/concepts/extend-kubernetes/operator/
- https://deploy-preview-14458--kubernetes-io-master-staging.netlify.com/docs/reference/glossary/?all=true#term-operator-pattern
- https://deploy-preview-14458--kubernetes-io-master-staging.netlify.com/docs/reference/kubectl/cheatsheet/

Reply:
> Check out the glossary reference, line 13, the link needs an opening paren?

Thanks for spotting this. Sorted.

> The An example Operator section is still a bit vague for me -- seems like pseudocode for writing an Operator. Could this content be combined with Writing your own Op?

I'm not sure I can improve on what I've written. If merging the PR makes the website clearly better with than without, I favor merging it, even if these changes fall short of perfect. That leaves room for some future PR to tweak the wording further.

> Do you need the link to the What's next section in line 108, that section is just below?

I don't mind linking to bits of page that are visible. I know some people aren't keen. That kind of link works well with a bit of CSS to highlight the fragment that got linked to (setting that effect up takes lots more work, though - it doesn't seem not worth doing now).

If this link needs to come out for the changes to land, I can do that.